x-pack/plugins/ml/public/application/explorer/anomalies_map.tsx

```diff
@@ -207,24 +204,26 @@ export const AnomaliesMap: FC<Props> = ({ anomalies, jobIds }) => {
     [...jobIds]
   );
 
-  const layerList = useMemo(() => {
-    let layers: VectorLayerDescriptor[] = [];
-    // Loop through suggestions list and make a layer for each
-    if (EMSSuggestions?.length) {
-      let count = 0;
-      layers = EMSSuggestions.reduce(function (result, suggestion) {
-        if (suggestion) {
-          const visible = count === 0;
-          result.push(getChoroplethAnomaliesLayer(anomalies, suggestion, visible));
-          count++;
-        }
-        return result;
-      }, [] as VectorLayerDescriptor[]);
-    }
-    return layers;
+  const layerList: VectorLayerDescriptor[] = useMemo(() => {
+    if (!EMSSuggestions?.length) return [];
+
+    return EMSSuggestions.map((suggestion) => {
+      return getChoroplethAnomaliesLayer(anomalies, suggestion);
+    }, [] as VectorLayerDescriptor[]);
   }, [EMSSuggestions, anomalies]);
 
-  if (EMSSuggestions?.length === 0) {
+  const layersWithAnomalies = layerList.filter((layer) => {
+    // @ts-ignore _rows does not exist - can remove when VectorLayerDescriptor is updated
+    const rows = layer.joins ? layer.joins[0].right.__rows : [];
```

Comment:
question: do we need a safaguard here? 
```suggestion
    const rows = Array.isArray(layer.joins) ? layer.joins[0]?.right?.__rows : [];
```

Reply:
Updated in 1a0d5ad9725dcb63cd403ff2fb1263f197e26982